xapian-core/cluster/cluster.cc

```diff
@@ -304,6 +311,11 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	double tf = 1 + log((double)wdf);
 	double idf = log(size / termfreq);
 	double wt = tf * idf;
+
+	// If weight of a term is zero, it doesn't contribute to the termlist
+	if (wt == 0)
+	    continue;
```

Comment:
I doubt there are many in that situation, so having code to remove those terms seems unnecessary. If it were to make a big difference, then something like PCA (which we believe will help more generally) would catch this.

Moving the check around as Olly has suggested will make the system faster.